src/nvim/eval/funcs.c

```diff
@@ -6824,13 +6824,43 @@ static void f_serverlist(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
 {
   size_t n;
   char **addrs = server_address_list(&n);
+  dict_T *addrs_dict = tv_dict_alloc();
 
   // Copy addrs into a linked list.
   list_T *const l = tv_list_alloc_ret(rettv, (ptrdiff_t)n);
   for (size_t i = 0; i < n; i++) {
     tv_list_append_allocated_string(l, addrs[i]);
+    tv_dict_add_bool(addrs_dict, addrs[i], strlen(addrs[i]), true);
   }
+
+  if (argvars[0].v_type != VAR_DICT) {
+    return;
+  }
+
+  if (!tv_dict_get_bool(argvars[0].vval.v_dict, "peer", false)) {
+    return;
+  }
+
+  Error err = ERROR_INIT;
+  Object rv = NLUA_EXEC_STATIC("return require('vim._core.server').serverlist(...)",
+                               (Array)ARRAY_DICT_INIT, kRetObject,
+                               NULL, &err);
+
+  if (ERROR_SET(&err)) {
+    ELOG("Error in function 'serverlist': %s", err.msg);
```

Comment:
do not write "error in" in errors, it's redundant.
```suggestion
    ELOG("vim._core.serverlist failed: %s", err.msg);
```